docs/guides/device-gateway/sdk.md

```diff
@@ -0,0 +1,344 @@
+
+![alt-text](img/dns-success.png)
+
+## Create a bot user
+
+First, you’ll create a bot user so that in the next section, you can create an agent authtoken independent of any user account.
+A bot user does not belong to a particular user account. Run the following command to create a new bot user, providing your API key and a description:
+
+```
+curl \
+-X POST \
+-H "Authorization: Bearer {API_KEY}" \
+-H "Content-Type: application/json" \
+-H "Ngrok-Version: 2" \
+-d '{"name":"new bot user from API"}' \
+https://api.ngrok.com/bot_users
+```
+
+You should receive a 201 response similar to the following:
+
+```json
+{
+	"id": "bot_2fmwUXhnImKswMgZKCsx7cnbt2l",
+	"uri": "https://api.ngrok.com/bot_users/bot_2fmwUXhnImKswMgZKCsx7cnbt2l",
+	"name": "new bot user from API",
+	"active": true,
+	"created_at": "2024-04-29T19:39:36Z"
+}
+```
+
+## Create the agent authtoken
+
+You should start each agent using a separate authtoken, and that token should belong to a bot user.
+
+To create an authtoken, login to the ngrok dashboard and click [**Authtokens**](https://dashboard.ngrok.com/tunnels/authtokens)
+under _Tunnels_, then click **Add a Tunnel Authtoken**.
+For **Owner**, select the bot user you use created, and select `bind:*.sitea.{YOUR_DOMAIN}` for the ACL Rules.
+This ACL will allow an agent with the authtoken to create tunnels on any subdomain of `sitea.{YOUR_DOMAIN}`.
+
+![alt-text](img/authtoken.png)
+
+## Configure the sample agent
+
+The sample application uses the ngrok Python SDK to embed connectivity directly into the application
+rather than using the standalone ngrok agent, the CLI, or the ngrok Operator. It serves a local REST
+API you can use to create, delete, and list tunnels.
+
+### Clone the repo
+
+Clone [this sample application](https://github.com/ngrok-samples/ngrok-iot-agent) on the device or device gateway you wish to connect to.
+
+### Create and activate a virtual environment
+
+Run the following to create a virtual environment, providing a directory for `{PATH_TO_VENV}`:
+
+`python -m venv {PATH_TO_VENV}`
+
+Run the appropriate command for your O/S to activate the virtual environment:
+
+- In cmd.exe
+  `{PATH_TO_VENV}\Scripts\activate.bat`
+- In PowerShell
+  `{PATH_TO_VENV}\Scripts\Activate.ps1`
+- Linux and MacOS
+  `source {PATH_TO_VENV}/bin/activate`
+
+### Install dependencies
+
+Run `pip install -r requirements` to install the dependencies.
+
+### Update the domain
+
+Update line `109` with your domain. Since you created a wildcard domain, you can use any subdomain
+of your wildcard domain without reserving it. For example, for the wildcard domain `*.sitea.configurable-domain.com`, you could use `api.sitea.configurable-domain.com`.
+
+### Set your authtoken
+
+This sample application is designed to read your ngrok credentials from the environment, so run the
+following command, replacing `{YOUR_AUTHTOKEN}` with the authtoken you created in the previous
+step:
+
+`export NGROK_AUTHTOKEN={YOUR_AUTHTOKEN}`
```

Comment:
Let's make that a code block.

````suggestion
```bash
export NGROK_AUTHTOKEN={YOUR_AUTHTOKEN}
```
````